content/courses/native-onchain-development/program-security.md

```diff
@@ -138,54 +150,66 @@ if pda != *note_pda.key {
 
 #### Ownership checks
 
-An ownership check verifies that an account is owned by the expected public key.
-Let's use the note-taking app example that we've referenced in previous lessons.
+An ownership check verifies that an account is owned by the expected program.
+For instance, if your program relies on PDAs (Program Derived Addresses), you
+want to ensure that those PDAs are controlled by your program and not by an
+external party.
+
+Let's use the note-taking app example that we've referenced in previous lessons
+[deserialize instruction data](/content/courses/native-onchain-development/deserialize-instruction-data.md)
+and
+[program state management](/content/courses/native-onchain-development/program-state-management.md).
 In this app, users can create, update, and delete notes that are stored by the
 program in PDA accounts.
 
-When a user invokes the `update` instruction, they also provide a `pda_account`.
-We presume the provided `pda_account` is for the particular note they want to
-update, but the user can input any instruction data they want. They could even
-potentially send data which matches the data format of a note account but was
-not also created by the note-taking program. This security vulnerability is one
-potential way to introduce malicious code.
+When a user invokes the `update` instruction handler, they also provide a
+`pda_account`. We presume the provided `pda_account` is for the particular note
+they want to update, but the user can input any instruction data they want. They
+could even potentially send data that matches the data format of a note account
+but was not also created by the note-taking program. This security vulnerability
+is one potential way to introduce malicious code.
 
 The simplest way to avoid this problem is to always check that the owner of an
 account is the public key you expect it to be. In this case, we expect the note
 account to be a PDA account owned by the program itself. When this is not the
 case, we can report it as an error accordingly.
 
+Here's how you can perform an ownership check to verify that an account is owned
+by the program:
+
 ```rust
 if note_pda.owner != program_id {
     return Err(ProgramError::InvalidNoteAccount);
 }
 ```
 
-As a side note, using PDAs whenever possible is more secure than trusting
-externally-owned accounts, even if they are owned by the transaction signer. The
-only accounts that the program has complete control over are PDA accounts,
-making them the most secure.
+In this example, we check if the `note_pda` is owned by the program itself
+(denoted by `program_id`). Ownership checks like these prevent unauthorized
+entities from tampering with critical accounts.
+
+<Callout>
 
-#### Signer checks
+PDAs, in particular, are more secure than externally-owned accounts because the
+program has full control over them. Ensuring ownership is a fundamental way to
+prevent malicious behavior. </Callout>
```

Comment:
```suggestion
PDAs are often considered to be trusted stores of a program's state. Ensuring the correct program owns the PDAs is a fundamental way to prevent malicious behavior. </Callout>
```

Reply:
Making it simpler...

Reply:
Updated as per suggestion.